code/game/machinery/cell_charger.dm

```diff
@@ -126,8 +126,11 @@
 /obj/machinery/cell_charger/RefreshParts()
 	. = ..()
 	charge_rate = 0.25 * STANDARD_CELL_RATE
-	for(var/datum/stock_part/capacitor/capacitor in component_parts)
-		charge_rate *= capacitor.tier
+
+	for(var/stock_part in component_parts)
+		if(istype(stock_part, /datum/stock_part/capacitor))
```

Comment:
The 1st format `for(var/datum/stock_part/capacitor/capacitor in component_parts)` is much cleaner compared to the one you have written below. Use `istype()` only if you have multiple types to check for and not a single one

Reply:
I am aware of this and I disagree. 

This makes it so that adding new functionality to stock parts in the future is easy, instead of having to rewrite each for loop. Leaving it as is, also increases the risk of someone reusing multiple for loops again, which is exactly what caused this whole mess. 

Monkey see, monkey do.

Reply:
> This makes it so that adding new functionality to stock parts in the future is easy

When the need arises in the future then we can use your approach but not here, not now. Leaving it in would teach others there are 2 ways to go about it.

> Leaving it as is, also increases the risk of someone reusing multiple for loops again

Not a good argument at all. It's something that can be caught in a code review, the risk factor is over rated

Reply:
Nobody corrected this approach in the 126 times it was used.

![Code_0eAewXHXRV](https://github.com/user-attachments/assets/531ed818-2cfe-4e3d-9ebf-e473da81759f)

If you leave it in, the same thing is _guaranteed_ to keep happening. People LOVE to c/p shit and reuse it without knowing how it works or what it does. It's the path of least resistance.

If you want a good example of this happening I present to you [BIO armor](https://github.com/tgstation/tgstation/pull/66721).